marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -912,7 +894,7 @@ arbitraryContractWeighted ((wClose, wPay, wIf, wWhen, wLet, wAssert) : w) contex
       (wClose , pure Close)
     , (wPay   , Pay <$> semiArbitrary context <*> semiArbitrary context <*> semiArbitrary context <*> semiArbitrary context <*> arbitraryContractWeighted w context)
     , (wIf    , If <$> semiArbitrary context <*> arbitraryContractWeighted w context <*> arbitraryContractWeighted w context)
-    , (wWhen  , When <$> listOf (arbitraryCaseWeighted w context) `suchThat` ((<= length w) . length) <*> semiArbitrary context <*> arbitraryContractWeighted w context)
+    , (wWhen  , When <$> (take (length w) <$> listOf (arbitraryCaseWeighted w context)) <*> semiArbitrary context <*> arbitraryContractWeighted w context)
```

Comment:
Don't use `suchThat` when it can be avoided. This is another example of the same problem and is almost certainly the reason why contract generation performance was affected by the size parameter to the degree it was.